enzyme/test/Enzyme/ForwardModeVector/bitcast.ll

```diff
@@ -0,0 +1,24 @@
+; RUN: %opt < %s %loadEnzyme -enzyme -enzyme-preopt=false -mem2reg -simplifycfg -adce -instsimplify -S | FileCheck %s
```

Comment:
Per above, if the applyChainRule is removed and the differential load/store is done directly, can we still get this without instsimplify?